shadowsocks/tcp.go

```diff
@@ -148,7 +149,7 @@ type TCPService interface {
 }
 
 // proxyConnection will route the clientConn according to the address read from the connection.
-func proxyConnection(clientConn onet.DuplexConn, proxyMetrics *metrics.ProxyMetrics) *onet.ConnectionError {
+func proxyConnection(clientConn onet.DuplexConn, proxyMetrics *metrics.ProxyMetrics, allowAllIPs bool) *onet.ConnectionError {
```

Comment:
Let's generalize this. Instead of passing allowAllIPs, pass a `isAllowedIP()` function that takes an IP address. Then we'll be able to share the ip validation logic across TCP and UDP, which is currently duplicated. In the test code, you can pass an allow all function.